cxx-qt-gen/src/parser/mod.rs

```diff
@@ -0,0 +1,169 @@
+                for item in items.1.drain(..) {
+                    // Try to find any CXX-Qt items, if found add them to the relevant
+                    // qobject. Otherwise return them to be added to other
+                    if let Some(other) = find_and_merge_cxx_qt_item(item, &mut cxx_qt_data)? {
+                        // Unknown item so add to the other list
+                        others.push(other);
+                    }
+                }
+            } else {
+                // No qobjects found so pass everything through
+                others.extend(items.1);
+            }
+        }
+
+        // Create a new module using only items that are not CXX-Qt items
+        module.content = Some((Brace::default(), others));
+
+        // Return the successful Parser object
+        Ok(Self {
+            passthrough_module: module,
+            cxx_qt_data,
+        })
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    use quote::{quote, ToTokens};
+    use syn::ItemMod;
+
+    /// Helper to parse a quote TokenStream into a given syn item
+    pub fn tokens_to_syn<T: syn::parse::Parse>(tokens: proc_macro2::TokenStream) -> T {
+        syn::parse2(tokens.into_token_stream()).unwrap()
+    }
+
+    #[test]
+    fn test_parser_from_empty_module() {
+        let module: ItemMod = tokens_to_syn(quote! {
+            #[cxx_qt::bridge]
+            mod ffi {}
+        });
+        let parser = Parser::from(module);
+        assert!(parser.is_ok());
+        let parser = parser.unwrap();
+        assert_eq!(parser.passthrough_module, module);
+        assert_eq!(parser.cxx_qt_data.qobjects.len(), 0);
+    }
+
+    #[test]
+    fn test_parser_from_cxx_items() {
+        let module: ItemMod = tokens_to_syn(quote! {
+            #[cxx_qt::bridge]
+            mod ffi {
+                extern "Rust" {
+                    fn test();
+                }
+            }
+        });
+        let parser = Parser::from(module);
+        assert!(parser.is_ok());
+        let parser = parser.unwrap();
+        assert_eq!(parser.passthrough_module, module);
+        assert_eq!(parser.cxx_qt_data.qobjects.len(), 0);
+    }
+
+    #[test]
+    fn test_parser_from_cxx_qt_items() {
+        let module: ItemMod = tokens_to_syn(quote! {
+            #[cxx_qt::bridge]
+            mod ffi {
+                struct RustObj;
+
+                #[cxx_qt::signals(RustObj)]
+                enum MySignals {
+                    Ready,
+                }
+            }
+        });
+        let parser = Parser::from(module);
+        assert!(parser.is_ok());
+        let parser = parser.unwrap();
+        assert_ne!(parser.passthrough_module, module);
+        assert_eq!(parser.passthrough_module.attrs.len(), 1);
+        assert_eq!(parser.passthrough_module.ident, "ffi");
+        assert_eq!(parser.passthrough_module.content.unwrap().1.len(), 0);
+        assert_eq!(parser.cxx_qt_data.qobjects.len(), 1);
```

Comment:
```suggestion
        let parser = parser.unwrap();
        
        assert_ne!(parser.passthrough_module, module);
        
        assert_eq!(parser.passthrough_module.attrs.len(), 1);
        assert_eq!(parser.passthrough_module.ident, "ffi");
        assert_eq!(parser.passthrough_module.content.unwrap().1.len(), 0);
        assert_eq!(parser.cxx_qt_data.qobjects.len(), 1);
```

It's easy to miss that one of these lines is `assert_ne` whereas the rest are `assert_eq`. IMO `assert_neq` would have been a better name for the macro to avoid this, but alas, that's not what it is called.

Same applies to the following tests.

Reply:
Done